src/devices/cellular.c

```diff
@@ -0,0 +1,534 @@
+                                                   message_val->data);
+
+                        return false;
+                }
+
+                pr_debug_str_msg("[cell] Unexpected auth response: ",
+                                 sb->buffer);
+        }
+
+        pr_info("[cell] Unable to authenticate telemetry server.\r\n");
+        return false;
+}
+
+int cellular_init_connection(DeviceConfig *config)
+{
+	telemetry_info.active_since = 0;
+
+        LoggerConfig *loggerConfig = getWorkingLoggerConfig();
+        CellularConfig *cellCfg =
+                &(loggerConfig->ConnectivityConfigs.cellularConfig);
+        TelemetryConfig *telemetryConfig =
+                &(loggerConfig->ConnectivityConfigs.telemetryConfig);
+
+        /* This is sane since DeviceConfig is typedef'd as a serial_buffer */
+        struct serial_buffer *sb = (struct serial_buffer*) config;
+        sb->serial->tx_callback = cell_serial_tx_cb;
+        sb->serial->rx_callback = cell_serial_rx_cb;
+
+        const int init_status = cellular_init_modem(sb);
+        if (DEVICE_INIT_SUCCESS != init_status)
+                return DEVICE_INIT_FAIL;
+
+        /* If here, there is a modem attached.  Figure out what modem it is */
+        enum cellular_modem modem_type = probe_cellular_manuf(sb);
+        if (!get_methods(modem_type, &methods)) {
+                pr_error_int_msg("Unable to load methods for modem_type ",
+                                 modem_type);
+                return DEVICE_INIT_FAIL;
+        }
+
+        if (!methods->init_modem(sb, &cell_info)) {
+                telemetry_info.status = TELEMETRY_STATUS_MODEM_INIT_FAILED;
+                return DEVICE_INIT_FAIL;
+        }
+
+        /* Read SIM data.  Don't care if these fail */
+        methods->get_sim_info(sb, &cell_info);
+
+        /* Now register on the network */
+        if (!methods->register_on_network(sb, &cell_info)) {
+                telemetry_info.status =
+                        TELEMETRY_STATUS_CELL_REGISTRATION_FAILED;
+                return DEVICE_INIT_FAIL;
+        }
+
+        /* Read in Network data.  Don't care if this fails */
+        pr_info("[cell] Network registered\r\n");
+        methods->get_network_info(sb, &cell_info);
+
+        /* Now setup a GPRS + PDP connection */
+        if (!methods->setup_pdp(sb, &cell_info, cellCfg)) {
+                telemetry_info.status =
+                        TELEMETRY_STATUS_INTERNET_CONFIG_FAILED;
+                return DEVICE_INIT_FAIL;
+        }
+
+        /* Connect to RCL */
+        if(!methods->open_telem_connection(sb, &cell_info, &telemetry_info,
+                                           telemetryConfig)) {
+                telemetry_info.status =
+                        TELEMETRY_STATUS_SERVER_CONNECTION_FAILED;
+                print_registration_failure();
+                return DEVICE_INIT_FAIL;
+        }
+
+        /* Auth against RCL */
+        if (!auth_telem_stream(sb, telemetryConfig)){
```

Comment:
Precisely... it would show as an auth failure for a bit until the next retry, which assuming they were still out of range, would lead to a network connection failure and a correct status.  Probably not work correcting now.